### PR TITLE
Update release tool

### PR DESCRIPTION
**Pull request by fengzixu**
**What type of PR is this?**
/kind feature
**What this PR does / why we need it**:

After [this pr](https://github.com/kubernetes-csi/csi-release-tools/pull/117) was merged, we should also update the csi-test version for csi-driver-host-path.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by pohly** [comment]
We'll have to step through the full test log to determine where it fails with a non-zero exit code. The JUnit file doesn't record any failed test. I'm still out for another week, though.

Another comment: please re-do the release-tools update with `git subtree --squash`. We don't want to see the individual commits anymore.

**Comment by pohly** [comment]
Looks like appending `/v4` is broken:
```
HEAD is now at af2d21f Merge pull request #306 from msau42/docs
Mon Jan  4 07:32:42 UTC 2021 go1.15.5 /home/prow/go/src/github.com/kubernetes-csi/csi-test/v4$ git clean -fdx
Mon Jan  4 07:32:42 UTC 2021 go1.15.5 $ go test -c -o /home/prow/go/pkg/csiprow.HvvZaqQE7w/csi-sanity github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity
go: finding module for package github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity
go: downloading github.com/kubernetes-csi/csi-test v1.1.1
go: downloading github.com/kubernetes-csi/csi-test/v4 v4.0.2
go: downloading github.com/kubernetes-csi/csi-test v2.2.0+incompatible
go: found github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity in github.com/kubernetes-csi/csi-test/v4 v4.0.2
go: downloading github.com/onsi/ginkgo v1.11.0
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/container-storage-interface/spec v1.3.0
go: downloading google.golang.org/grpc v1.34.0
go: downloading github.com/onsi/gomega v1.7.1
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading github.com/golang/protobuf v1.4.3
go: downloading google.golang.org/protobuf v1.25.0
go: downloading golang.org/x/net v0.0.0-20201209123823-ac852fbbde11
go: downloading google.golang.org/genproto v0.0.0-20201209185603-f92720507ed4
go: downloading golang.org/x/text v0.3.4
go: downloading github.com/hpcloud/tail v1.0.0
go: downloading golang.org/x/sys v0.0.0-20201207223542-d4d67f95c62d
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: downloading gopkg.in/fsnotify.v1 v1.4.7
?   	github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity	[no test files]
Mon Jan  4 07:33:02 UTC 2021 go1.15.5 $ /home/prow/go/pkg/csiprow.HvvZaqQE7w/csi-sanity -ginkgo.v -csi.junitfile /logs/artifacts/junit_sanity.xml -csi.endpoint dns:///172.18.0.2:30489 -csi.stagingdir /tmp/staging -csi.mountdir /tmp/mount -csi.createstagingpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/mkdir_in_pod.sh -csi.createmountpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/mkdir_in_pod.sh -csi.removestagingpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/rmdir_in_pod.sh -csi.removemountpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/rmdir_in_pod.sh
release-tools/prow.sh: line 361: /home/prow/go/pkg/csiprow.HvvZaqQE7w/csi-sanity: No such file or directory
```

That change had been tested (https://github.com/kubernetes-csi/csi-release-tools/pull/117#discussion_r532694862 -> https://github.com/kubernetes-csi/csi-driver-host-path/pull/210) but something must have gone wrong when testing that PR because the prow.sh change is broken without any obvious test failure.

@fengzixu Please fix the issue in this PR and then port it back to csi-release-tools. I think you can simply use `CSI_PROW_SANITY_IMPORT_PATH` as it was before. `PACKAGE_PATH` probably would be a better name for it. This code pre-dates the introduction of versioned package imports :cry:

**Comment by fengzixu** [comment]
@pohly I am not sure if my understanding is correct about your [comment](https://github.com/kubernetes-csi/csi-driver-host-path/pull/234#issuecomment-757821453). Please check this PR: https://github.com/kubernetes-csi/csi-driver-host-path/pull/234#issuecomment-757821453

**Comment by fengzixu** [comment]
> Looks like appending `/v4` is broken:
> 
> ```
> HEAD is now at af2d21f Merge pull request #306 from msau42/docs
> Mon Jan  4 07:32:42 UTC 2021 go1.15.5 /home/prow/go/src/github.com/kubernetes-csi/csi-test/v4$ git clean -fdx
> Mon Jan  4 07:32:42 UTC 2021 go1.15.5 $ go test -c -o /home/prow/go/pkg/csiprow.HvvZaqQE7w/csi-sanity github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity
> go: finding module for package github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity
> go: downloading github.com/kubernetes-csi/csi-test v1.1.1
> go: downloading github.com/kubernetes-csi/csi-test/v4 v4.0.2
> go: downloading github.com/kubernetes-csi/csi-test v2.2.0+incompatible
> go: found github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity in github.com/kubernetes-csi/csi-test/v4 v4.0.2
> go: downloading github.com/onsi/ginkgo v1.11.0
> go: downloading github.com/google/uuid v1.1.2
> go: downloading github.com/container-storage-interface/spec v1.3.0
> go: downloading google.golang.org/grpc v1.34.0
> go: downloading github.com/onsi/gomega v1.7.1
> go: downloading gopkg.in/yaml.v2 v2.2.8
> go: downloading github.com/golang/protobuf v1.4.3
> go: downloading google.golang.org/protobuf v1.25.0
> go: downloading golang.org/x/net v0.0.0-20201209123823-ac852fbbde11
> go: downloading google.golang.org/genproto v0.0.0-20201209185603-f92720507ed4
> go: downloading golang.org/x/text v0.3.4
> go: downloading github.com/hpcloud/tail v1.0.0
> go: downloading golang.org/x/sys v0.0.0-20201207223542-d4d67f95c62d
> go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
> go: downloading gopkg.in/fsnotify.v1 v1.4.7
> ?   	github.com/kubernetes-csi/csi-test/v4/cmd/csi-sanity	[no test files]
> Mon Jan  4 07:33:02 UTC 2021 go1.15.5 $ /home/prow/go/pkg/csiprow.HvvZaqQE7w/csi-sanity -ginkgo.v -csi.junitfile /logs/artifacts/junit_sanity.xml -csi.endpoint dns:///172.18.0.2:30489 -csi.stagingdir /tmp/staging -csi.mountdir /tmp/mount -csi.createstagingpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/mkdir_in_pod.sh -csi.createmountpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/mkdir_in_pod.sh -csi.removestagingpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/rmdir_in_pod.sh -csi.removemountpathcmd /home/prow/go/pkg/csiprow.HvvZaqQE7w/rmdir_in_pod.sh
> release-tools/prow.sh: line 361: /home/prow/go/pkg/csiprow.HvvZaqQE7w/csi-sanity: No such file or directory
> ```
> 
> That change had been tested ([kubernetes-csi/csi-release-tools#117 (comment)](https://github.com/kubernetes-csi/csi-release-tools/pull/117#discussion_r532694862) -> #210) but something must have gone wrong when testing that PR because the prow.sh change is broken without any obvious test failure.
> 
> @fengzixu Please fix the issue in this PR and then port it back to csi-release-tools. I think you can simply use `CSI_PROW_SANITY_IMPORT_PATH` as it was before. `PACKAGE_PATH` probably would be a better name for it. This code pre-dates the introduction of versioned package imports 😢

I think if I didn't add `/v4`, the ci test of [this PR](https://github.com/kubernetes-csi/csi-driver-host-path/pull/210) will not pass. Because even if we checkout csi-test to v4.0.2 version, when we run `go test` command, it still fetch csi-test v2.2.x

```
Sat Jan 16 12:04:47 UTC 2021 go1.15.5 $ git init /home/prow/go/src/github.com/kubernetes-csi/csi-test
Initialized empty Git repository in /home/prow/go/src/github.com/kubernetes-csi/csi-test/.git/
Sat Jan 16 12:04:47 UTC 2021 go1.15.5 /home/prow/go/src/github.com/kubernetes-csi/csi-test$ git fetch --depth=1 https://github.com/kubernetes-csi/csi-test v4.0.2
From https://github.com/kubernetes-csi/csi-test
 * tag               v4.0.2     -> FETCH_HEAD
Sat Jan 16 12:04:48 UTC 2021 go1.15.5 /home/prow/go/src/github.com/kubernetes-csi/csi-test$ git checkout FETCH_HEAD
Note: checking out 'FETCH_HEAD'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at af2d21f Merge pull request #306 from msau42/docs
Sat Jan 16 12:04:48 UTC 2021 go1.15.5 /home/prow/go/src/github.com/kubernetes-csi/csi-test$ git clean -fdx
Sat Jan 16 12:04:48 UTC 2021 go1.15.5 $ go test -c -o /home/prow/go/pkg/csiprow.1UdCqheJ9R/csi-sanity github.com/kubernetes-csi/csi-test/cmd/csi-sanity
go: finding module for package github.com/kubernetes-csi/csi-test/cmd/csi-sanity
go: downloading github.com/kubernetes-csi/csi-test v1.1.1
go: downloading github.com/kubernetes-csi/csi-test v2.2.0+incompatible
go: found github.com/kubernetes-csi/csi-test/cmd/csi-sanity in github.com/kubernetes-csi/csi-test v2.2.0+incompatible
go: downloading github.com/onsi/ginkgo v1.11.0
go: downloading github.com/onsi/gomega v1.7.0
go: downloading google.golang.org/grpc v1.34.0
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/container-storage-interface/spec v1.3.0
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/hpcloud/tail v1.0.0
go: downloading golang.org/x/net v0.0.0-20201209123823-ac852fbbde11
go: downloading google.golang.org/protobuf v1.25.0
go: downloading google.golang.org/genproto v0.0.0-20201209185603-f92720507ed4
go: downloading golang.org/x/text v0.3.4
go: downloading gopkg.in/fsnotify.v1 v1.4.7
go: downloading golang.org/x/sys v0.0.0-20201207223542-d4d67f95c62d
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
Sat Jan 16 12:05:02 UTC 2021 go1.15.5 $ /home/prow/go/pkg/csiprow.1UdCqheJ9R/csi-sanity -ginkgo.v -csi.junitfile /logs/artifacts/junit_sanity.xml -csi.endpoint dns:///172.18.0.4:30572 -csi.stagingdir /tmp/staging -csi.mountdir /tmp/mount -csi.createstagingpathcmd /home/prow/go/pkg/csiprow.1UdCqheJ9R/mkdir_in_pod.sh -csi.createmountpathcmd /home/prow/go/pkg/csiprow.1UdCqheJ9R/mkdir_in_pod.sh -csi.removestagingpathcmd /home/prow/go/pkg/csiprow.1UdCqheJ9R/rmdir_in_pod.sh -csi.removemountpathcmd /home/prow/go/pkg/csiprow.1UdCqheJ9R/rmdir_in_pod.sh
```
@pohly

**Comment by pohly** [comment]
Instead of `go test -c -o /home/prow/go/pkg/csiprow.1UdCqheJ9R/csi-sanity github.com/kubernetes-csi/csi-test/cmd/csi-sanity` we have to use `go test -c -o /home/prow/go/pkg/csiprow.1UdCqheJ9R/csi-sanity ./cmd/csi-sanity` and run that command in the directory where we checked out csi-test.

**Comment by fengzixu** [comment]
@pohly I two issues in prow.sh script is

1. there is no test file under /csi-test/cmd/csi-sanity in v4.0.2 version. It was old code in v2.2.0. In that case, we should use go test -c to build binary file of csi-sanity  https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/release-tools/prow.sh#L883
2. fetch csi-test from upstream always

In that case, I modified the logic in prow.sh, we need to build binary file of csi-sanity and move it to the target directory

```bash
git_checkout "${CSI_PROW_SANITY_REPO}" "${GOPATH}/src/${CSI_PROW_SANITY_PACKAGE_PATH}" "${CSI_PROW_SANITY_VERSION}" --depth=1 || die "checking out csi-sanity failed"
cd "${GOPATH}/src/${CSI_PROW_SANITY_PACKAGE_PATH}/cmd/csi-sanity" && go build || die "building csi-sanity failed"
mv ./csi-sanity "${CSI_PROW_WORK}"
```

File: release-tools/prow.sh [hunk 1]
```diff
@@ -879,8 +879,8 @@ install_sanity () (
         return
     fi
 
-    git_checkout "${CSI_PROW_SANITY_REPO}" "${GOPATH}/src/${CSI_PROW_SANITY_IMPORT_PATH}" "${CSI_PROW_SANITY_VERSION}" --depth=1 || die "checking out csi-sanity failed"
-    run_with_go "${CSI_PROW_GO_VERSION_SANITY}" go test -c -o "${CSI_PROW_WORK}/csi-sanity" "${CSI_PROW_SANITY_IMPORT_PATH}/cmd/csi-sanity" || die "building csi-sanity failed"
+    git_checkout "${CSI_PROW_SANITY_REPO}" "${GOPATH}/src/${CSI_PROW_SANITY_PACKAGE_PATH}" "${CSI_PROW_SANITY_VERSION}" --depth=1 || die "checking out csi-sanity failed"
+    run_with_go "${CSI_PROW_GO_VERSION_SANITY}" go test -c -o "${CSI_PROW_WORK}/csi-sanity" "${CSI_PROW_SANITY_PACKAGE_PATH}/cmd/csi-sanity" || die "building csi-sanity failed"
```
Comment (pohly) [inline release-tools/prow.sh:883]: From https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_csi-driver-host-path/234/pull-kubernetes-csi-csi-driver-host-path-1-17-on-kubernetes-1-17/1350407805189754880/build-log.txt:

```
Sat Jan 16 11:45:07 UTC 2021 go1.15.5 $ go test -c -o /home/prow/go/pkg/csiprow.IW3LnBgFPK/csi-sanity github.com/kubernetes-csi/csi-test/cmd/csi-sanity
go: finding module for package github.com/kubernetes-csi/csi-test/cmd/csi-sanity
go: downloading github.com/kubernetes-csi/csi-test v1.1.1
go: downloading github.com/kubernetes-csi/csi-test v2.2.0+incompatible
go: found github.com/kubernetes-csi/csi-test/cmd/csi-sanity in github.com/kubernetes-csi/csi-test v2.2.0+incompatible
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading google.golang.org/grpc v1.34.0
go: downloading github.com/onsi/ginkgo v1.11.0
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/onsi/gomega v1.7.0
go: downloading github.com/container-storage-interface/spec v1.3.0
go: downloading golang.org/x/net v0.0.0-20201209123823-ac852fbbde11
go: downloading github.com/golang/protobuf v1.4.3
go: downloading google.golang.org/protobuf v1.25.0
go: downloading github.com/hpcloud/tail v1.0.0
go: downloading gopkg.in/fsnotify.v1 v1.4.7
go: downloading golang.org/x/text v0.3.4
go: downloading google.golang.org/genproto v0.0.0-20201209185603-f92720507ed4
go: downloading golang.org/x/sys v0.0.0-20201207223542-d4d67f95c62d
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
```

I'm pretty sure that this ignored the source that was checked out earlier. What we need to do instead is check out the source, then inside that source code directory build with a relative package path:
```
go build -o "${CSI_PROW_WORK}/csi-sanity" ./cmd/csi-sanity
```

**Review by pohly — CHANGES_REQUESTED** [review]

File: release-tools/prow.sh (hunk 1 above)
Comment (pohly) [reply release-tools/prow.sh:883]: Sorry, I'm getting confused by the number of open PRs for this.

I see that you switched to `go build + mv` in https://github.com/kubernetes-csi/csi-driver-host-path/pull/210. A better solutution would be to use `go build -o <target path>`.

Comment (fengzixu) [reply release-tools/prow.sh:883]: https://github.com/kubernetes-csi/csi-driver-host-path/pull/210 is for testing if v4.0.2 can satisfy the requirement of new capability 

https://github.com/kubernetes-csi/csi-driver-host-path/pull/234 is for updating release-tool to host-path driver

But, now, we found a bug in csi-test, that's reason why I submit a new PR https://github.com/kubernetes-csi/csi-release-tools/pull/128

Those PRs should be merge based on the order as below

1. https://github.com/kubernetes-csi/csi-release-tools/pull/128
2. https://github.com/kubernetes-csi/csi-driver-host-path/pull/234
3. https://github.com/kubernetes-csi/csi-driver-host-path/pull/210

@pohly

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**